src/olympia/files/tasks.py

```diff
@@ -29,3 +37,51 @@ def extract_webext_permissions(ids, **kw):
                     defaults={'permissions': permissions}, file=file_)
         except Exception, err:
             log.error('Failed to extract: %s, error: %s' % (file_.pk, err))
+
+
+WEBEXTPERMS_DESCRIPTION_REGEX = r'^webextPerms\.description\.(.+)=(.+)'
+
+
+@task
+@write
+def update_webext_descriptions(url, locale='en-US', create=True, **kw):
+    class DummyContextManager(object):
+        def __enter__(self):
+            pass
+
+        def __exit__(*x):
+            pass
+
+    log.info('Updating webext permission descriptions in [%s] from %s' %
+             (locale, url))
+    try:
+        response = requests.get(url)
+    except RequestException as e:
```

Comment:
That only handles things like DNS error, timeouts etc. What about 40x or 50x errors ? One quick and simple way to implement it would be to use `response.raise_for_status()` in the try block.